FanGraphSpider.py

```diff
@@ -1,174 +0,0 @@
-}
-
-#Builds URLs to be scraped from team name and date. Team Name Gathered from MLB SPider.
-starturls = []
-dt = datetime.date.today().strftime('%Y-%m-%d')
-with open('schedule.csv', 'r') as input:
-    reader = csv.reader(input)
-    for item in reader:
-        # Only adds item to url if time is later than scheduled game time
-        if item[0] in teamdict.keys() and item[1] < datetime.datetime.now().strftime('%H:%M'):
-            starturls.append('https://www.fangraphs.com/livewins.aspx?date=%s&team=%s&dh=0&season=%s' % (dt, teamdict[item[0]], dt[:4]))
-
-
-class FanGraphSpider(scrapy.Spider):
-    name = 'FanGraphSpider'
-    start_urls = starturls
-    SPLASH_ARGS = {
-        'wait': 20.0,
-        'images': 0,
-        'render_all': 1,
-        'timeout': 40.0,
-        'resource_timeout': 40.0}
-
-    def parse(self, response):
-        # Checks for Home Runs and stolen bases in play log and, if found, sends player stat page to parser.
-        tweeted = False
-        for item in response.xpath('//div[@id="LiveGame1_dgPlay"]/table/tbody/tr'):
-            if 'homered' in item.xpath('td[6]/text()').extract_first():
-                # Checks to see if play has already been sent to parser. If not, adds it to list to avoid duplicates.
-                plname = item.xpath('td[2]/text()').extract_first()
-                with open('dupcatcher.csv', 'r') as input:
-                    reader = csv.reader(input)
-                    for line in reader:
-                        if [item.xpath('td[6]/text()').extract_first()] == line:
-                            tweeted = True
-                if not tweeted:
-                    with open('dupcatcher.csv', 'a') as output:
-                        writer = csv.writer(output)
-                        line = item.xpath('td[6]/text()').extract_first()
-                        writer.writerow([line])
-                    for namespot in response.xpath('//div[contains(@id, "dg3")]/table/tbody/tr/td[1] | //div[contains(@id, "dg4")]/table/tbody/tr/td[1]'):
-                        name = namespot.xpath('a/text()').extract_first()
-                        if plname == name:
-                            url = namespot.xpath('a/@href').extract_first()
-                            break
-                    url = response.urljoin(url)
-                    yield SplashRequest(url, callback=self.home_parse, args=dict(self.SPLASH_ARGS))
-            if 'stolen' in item.xpath('td[6]/text()').extract_first():
-                plname = item.xpath('td[2]/text()').extract_first()
-                # Checks to see if play has already been sent to parser. If not, adds it to list to avoid duplicates.
-                with open('dupcatcher.csv', 'r') as input:
-                    reader = csv.reader(input)
-                    for line in reader:
-                        if [item.xpath('td[6]/text()').extract_first()] == line:
-                            tweeted = True
-                if not tweeted:

```

Comment:
And having the `is_tweeted` function from above would let you get rid of this duplicate code down here.